atom/browser/api/atom_api_system_preferences_mac.mm

```diff
@@ -438,6 +443,59 @@ AVMediaType ParseMediaType(const std::string& media_type) {
   }
 }
 
+bool SystemPreferences::IsTouchIDAvailable() {
+  if (@available(macOS 10.12.2, *)) {
+    base::scoped_nsobject<LAContext> context([[LAContext alloc] init]);
+    return [context
+        canEvaluatePolicy:LAPolicyDeviceOwnerAuthenticationWithBiometrics
+                    error:nil];
+  }
+  return false;
+}
+
+void OnTouchIDCompleted(scoped_refptr<util::Promise> promise, bool success) {
+  promise->Resolve(success);
+}
+
+v8::Local<v8::Promise> SystemPreferences::PromptTouchID(
+    v8::Isolate* isolate,
+    const std::string& reason) {
+  scoped_refptr<util::Promise> promise = new util::Promise(isolate);
+  if (@available(macOS 10.12.2, *)) {
```

Comment:
Would it make sense to use `IsTouchIDAvailable()` here instead of just checking for 10.12.2? That way the other biometrics tests in that function will can be reused here and let us error out gracefully in this func's promise->RejectWithErrorMessage()`

Reply:
@codebytere points out this doesn't make sense because @available is like an `#ifdef` in that we need this at compile time to ensure use of the api tokens introduced in 10.12.2 :+1: